testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPBinaryAttributesTest.java

```diff
@@ -156,18 +158,16 @@ public void test03WritableMapper() {
         // Assert he is found including jpegPhoto
         joe = getUserAndAssertPhoto("joephoto", true);
 
+        // Assert that local storage doesn't contain LDAPConstants.JPEG_PHOTO, it should be stored in the LDAP
+        String joeId = joe.getId();
+        testingClient.server().run(session -> {
+            RealmModel test = session.realms().getRealmByName("test");
+            UserModel userById = session.userLocalStorage().getUserById(joeId, test);
+
+            assertThat(userById.getAttributes().get(LDAPConstants.JPEG_PHOTO), is(nullValue()));
+        });
 
-        // Try to update him with some big non-LDAP mapped attribute. It will fail
-        try {
-            joe.getAttributes().put("someOtherPhoto", Arrays.asList(JPEG_PHOTO_BASE64));
-            adminClient.realm("test").users().get(joe.getId()).update(joe);
-            Assert.fail("Not expected to successfully update user");
-        } catch (ClientErrorException cee) {
-            // Expected
-        }
```

Comment:
If I understand it correctly the test is actually changed to test something else than before, right? Can you please explain to me what has changed?

Reply:
This test is testing that attribute with name `LDAPConstants.JPEG_PHOTO` is not stored in local storage but in LDAP because of the mapper created on line 147. The original implementation was relying on the fact that the database is not even able to store such a long value > 255 character (probably depending on DB). This means that if storing of  `LDAPConstants.JPEG_PHOTO` attribute passed it was not stored in DB but in LDAP. However, if we try to store attribute `someOtherPhoto` with the same value, it will throw `ClientErrorException` because this time DB will refuse the value.

However, with Map storage we can store such values, therefore this test was not working correctly. I changed it so that we are not testing based on DB refusing the value, but we just check local storage for the attribute not to be present there.

Reply:
It makes perfect sense to me now, thank you for explanation.